packages/overlays/src/OverlayMixin.js

```diff
@@ -15,52 +16,110 @@ export const OverlayMixin = dedupeMixin(
             type: Boolean,
             reflect: true,
           },
-          popperConfig: Object,
+          config: {
+            type: Object,
+          },
+          closeEventName: {
+            type: String,
+          },
         };
       }
 
+      constructor() {
+        super();
+        this.config = {};
+        this.closeEventName = 'overlay-close';
+      }
+
       get opened() {
         return this._overlayCtrl.isShown;
       }
 
       set opened(show) {
+        if (show) {
+          this.dispatchEvent(new Event('before-show'));
```

Comment:
we already have that event in the OverlayController let's better reDispatch then to have another timing